src/m365/planner/commands/tenant/tenant-settings-set.ts

```diff
@@ -0,0 +1,125 @@
+import { Logger } from '../../../../cli';
+import { CommandOption } from '../../../../Command';
+import { AxiosRequestConfig } from 'axios';
+import { validation } from '../../../../utils';
+import request from '../../../../request';
+import PlannerCommand from '../../../base/PlannerCommand';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  isPlannerAllowed?: boolean;
+  allowCalendarSharing?: boolean;
+  allowTenantMoveWithDataLoss?: boolean;
+  allowTenantMoveWithDataMigration?: boolean;
+  allowRosterCreation?: boolean;
+  allowPlannerMobilePushNotifications?: boolean;
+}
+
+class PlannerTenantSettingsSetCommand extends PlannerCommand {
+  public get name(): string {
+    return commands.TENANT_SETTINGS_SET;
+  }
+
+  public get description(): string {
+    return 'Sets Microsoft Planner configuration of the tenant';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.isPlannerAllowed = typeof args.options.isPlannerAllowed !== 'undefined';
+    telemetryProps.allowCalendarSharing = typeof args.options.allowCalendarSharing !== 'undefined';
+    telemetryProps.allowTenantMoveWithDataLoss = typeof args.options.allowTenantMoveWithDataLoss !== 'undefined';
+    telemetryProps.allowTenantMoveWithDataMigration = typeof args.options.allowTenantMoveWithDataMigration !== 'undefined';
+    telemetryProps.allowRosterCreation = typeof args.options.allowRosterCreation !== 'undefined';
+    telemetryProps.allowPlannerMobilePushNotifications = typeof args.options.allowPlannerMobilePushNotifications !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['isPlannerAllowed', 'allowCalendarSharing', 'allowTenantMoveWithDataLoss', 'allowTenantMoveWithDataMigration', 'allowRosterCreation', 'allowPlannerMobilePushNotifications'];
+  }
```

Comment:
This section is not required since we do not have a command returning an array of values